File: repository/src/ch/uzh/ifi/feedback/repository/model/Category.java
package ch.uzh.ifi.feedback.repository.model;

import ch.uzh.ifi.feedback.library.rest.annotations.DbAttribute;
import ch.uzh.ifi.feedback.library.rest.validation.Id;

public class Category {

	@Id
	private long id;
	
	private String text;
	
	@DbAttribute("category_feedback_id")
	private transient Integer categoryFeedbackId;
	
	private CategoryType categoryType;	
}

File: reporting_frontend/src/client/app/shared/models/feedbacks/category.ts
import {CategoryType} from './category_type';


export class Category {
    mechanismId:number;
    text:string;
    categoryType:CategoryType;

    constructor(mechanismId:number, text:string, categoryTypeId:number, categoryType:CategoryType) {
        this.mechanismId = mechanismId;
        this.text = text;
        this.categoryType = categoryType;
    }
}

